### ISSUE TITLE
Please stop using "var" as a general type declaration

### ISSUE THREAD

**Issue by Paul-5** [dotnet/AspNetCore.Docs#29557]
Could people please stop using "var" as a type declaration.

The original intended use of "var" is for anonymous functions. Using "var" in general code, particularly in examples in forums, makes the code extremely hard to follow as you can't determine what types are being returned from some of the method calls.

I'm currently trying to research a topic that's new to me but all the method calls are typed as "var" so I don't know what framework classes to research... it's very frustrating.

Thank you.

**Comment by Rick-Anderson** [comment]
> Could people please stop using "var" as a type declaration.
> 
> The original intended use of "var" is for anonymous functions. Using "var" in general code, particularly in examples in forums, makes the code extremely hard to follow as you can't determine what types are being returned from some of the method calls.
> 
> I'm currently trying to research a topic that's new to me but all the method calls are typed as "var" so I don't know what framework classes to research... it's very frustrating.
> 
> Thank you.

Download the sample and VS shows the type. You can then debug the code, modify the code, etc.

**Comment by mkArtakMSFT** [comment]
Closing as this is not something we plan to change.

**Comment by Paul-5** [comment]
Thank you Rick,

 

If I’m researching an issue on an Android or iPhone device then your suggestion will not work.

 

In addition, having to copy little snippets of code out of GitHub etc. into VS to then embed them into a class and include the correct packages, references, etc. just to find out the type is impractical.

 

Unfortunately the detrimental effects of using var weren’t fully considered when “Usage of the var keyword” was written in the .Net Core Engineering Guidelines. There are even logical flaws in the examples they provide, i.e. they make it clear that the ability to use var is dependent upon the value of the variable… that’s just silliness – set the variable to the correct type in the first place.

 

In relation to GitHub, if code snippets cannot be provided in the clearest possible form the benefit of providing the snippets is partially lost.

 

Regards,

 

Paul.

 

_________________________________________
Paul Hopkins - Director
Senior IT Solution Architect and Developer
 <tel:0417043977> 0417 043 977 |  ***@***.***> ***@***.***

Atom Software Pty Ltd
 <https://www.atomsoftware.com.au/> https://www.atomsoftware.com.au
